spec/osut_tests_spec.rb

```diff
@@ -1333,1134 +1129,2447 @@ module M
+    sub = {}
+    sub[:id    ] = "South Skylight"
+    sub[:type  ] = "Skylight"
+    sub[:height] = side
+    sub[:width ] = side
+    sub[:head  ] = head
+    sub[:count ] = 2
+    sub[:offset] = offset
+    expect(mod1.addSubs(model, attic_south, [sub])).to be(true)
+    expect(mod1.status.zero?).to be(true)
+
+    # Re-validating pre-tested areas + volumes, as well as convexity.
+    expect(core.floorArea).to be_within(TOL).of(149.66)
+    core_volume = core.floorArea * core.ceilingHeight
+    expect(core_volume).to be_within(TOL).of(core.volume)
+    expect(attic.volume).to be_within(TOL).of(720.19)
+    expect(attic.floorArea).to be_within(TOL).of(567.98)
+
+    expect(mod1.poly(core_floor, true).empty?).to be(true)   # now concave
+    expect(mod1.poly(core_ceiling, true).empty?).to be(true) # now concave
+    expect(mod1.poly(attic_floor, true).empty?).to be(true)  # now concave
+    expect(mod1.logs.size).to eq(3)
+    expect(mod1.error?).to be(true)
+    mod1.logs.each {|l| expect(l[:message].include?("non-convex")).to be(true)}
+    expect(mod1.clean!).to eq(DBG)
+
+    shd = model.getShadowCalculation
+    # shadingCalculationMethodValues:
+    #   - "PolygonClipping"                       # (default)
+    #   - "PixelCounting"                         # allows concave, yet **
+    #
+    #   ** github.com/NREL/EnergyPlus/issues/9059 (no transparent shading yet)
+    #
+    # polygonClippingAlgorithmValues:
+    #   - "ConvexWeilerAthertonSutherlandHodgman" # convex only
+    #   - "SlaterBarskyand"                       # rectangular only
+    #   - "SutherlandHodgman"                     # (default)
+    expect(shd.shadingCalculationMethod).to eq("PolygonClipping")
+    expect(shd.polygonClippingAlgorithm).to eq("SutherlandHodgman")
+
+    ctl = model.getSimulationControl
+    # validSolarDistributionValues:
+    #   - "MinimalShadowing"
+    #   - "FullExterior"
+    #   - "FullInteriorAndExterior"
+    #   - "FullExteriorWithReflections"
+    #   - "FullInteriorAndExteriorWithReflections"
+    expect(ctl.solarDistribution).to eq("FullInteriorAndExterior")
+    expect(ctl.isSolarDistributionDefaulted).to be(false)
+    expect(ctl.shadowCalculation.get).to eq(shd)
+
+    file = File.join(__dir__, "files/osms/out/mini.osm")
+    model.save(file, true)
+    # Simulation results E+ 22.2 (SutherlandHodgman, PolygonClipping)
+    # ... no E+ errors/warnings.
+    #
+    # office.osm  FullInteriorAndExterior                 247 GJ 3.8 UMH cool
+    # mini.osm A  FullInteriorAndExterior                 247 GJ 4.0 UMH cool
+    # mini.osm B  FullExteriorWithReflections             248 GJ 3.8 UMH cool
+    # mini.osm C  FullInteriorAndExteriorWithReflections  247 GJ 3.8 UMH cool
+
+    # --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- #
+    # Mini as an interior courtyard:
+    #   - 4x new outdoor-facing walls (core)
+    #   - remove mini floor @Z0 and mini ceiling @Z3.05
+    #   - attic mini floor facing outdoors
```

Comment:
This transforms the core space into a _donut_, something considered [problematic](https://unmethours.com/question/33919/zone-within-a-zone/?answer=33925#post-id-33925) for OpenStudio. Not sure if this is strictly a fixed issue as of v350, but the area and volume calculations remain consistent, as well as E+ 22.2 simulation results.